app/jobs/create_institution_records_job.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'net/http'
+
+class CreateInstitutionRecordsJob < ApplicationJob
+  queue_as :default
+
+  def perform(institution_type_id)
+    i = 1
+    loop do
+      rspo_api_uri = URI("http://194.54.26.132/api/placowki/?page=#{i}&typ_podmiotu_id=#{institution_type_id}")
```

Comment:
Have you ever heard of clients & wrappers? They would fit perfectly here: https://medium.com/selleo/essential-rubyonrails-patterns-clients-and-wrappers-c19320bcda0
That will allow you to reuse api-related logic throughout the application in an elegant way, and it will make it way easier to test this code

Also, for testing, you can use https://github.com/bblimke/webmock or https://github.com/vcr/vcr

Reply:
I heard of them before but I never had the opportunity do design one. I don't think it is particularly necessary here because we are making only two distinct calls to this api and we are unlikely to need more in the near future.

Reply:
As long as you can write good tests for that without introducing a client, I'm okay with that - although IMO clients are pretty easy to write, and they make it way easier to test the integration